src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -1992,6 +2054,117 @@ public void ChainTHasCheckpointAt1000_MaxReorgIs10_TwoChainsPriorTo1000Presented
+
+            // Additional SetUp for current test.
+            ChainedHeader chainKTip = chainDTip; //peer K has exactly the same chain as peer D.
+            List<BlockHeader> peerKBlockHeaders = ctx.ChainedHeaderToList(chainKTip, chainKTip.Height);
+
+            // Claiming Peer K chain.
+            ConnectNewHeadersResult connectionPeerKResult = cht.ConnectNewHeaders(4, peerKBlockHeaders);
+            ctx.SwitchToChain(cht, chainKTip, connectionPeerKResult.Consumed, 5);
+
+            var chainedHeadersWithPeerK = new Dictionary<uint256, ChainedHeader>(cht.GetChainedHeadersByHash());
+
+            cht.PeerDisconnected(4);
+
+            Dictionary<uint256, ChainedHeader> chainedHeadersWithoutPeerK = cht.GetChainedHeadersByHash();
+
+            chainedHeadersWithoutPeerK.Should().BeEquivalentTo(chainedHeadersWithPeerK);
+
+            this.CheckChainedHeaderTreeConsistency(cht, listOfAllUniqueBlockHeaders);
+        }
+
+        /// <summary>
+        /// Basic test for tests 18 to 20, 28.
+        /// 1/6 There is no branch which is claimed by no one (PeerTipsByHash have tip of every branch), none of list should be empty.
+        /// 2/6 ChainHeadersByHash contains only reachable headers.
+        /// 3/6 ChainHeader claims LocalPeer.
+        /// 4/6 Each connected peer has exactly 1 entry in PeertipsByHash.
+        /// 5/6 CH.Next[i].Prev == CH for every header and every .Next.
+        /// 6/6 PeerTipsByPeerId should reflect PeerIdsByHash(except local marker).
+        /// </summary>
+        /// <param name="cht">ChainHeaderTree.</param>
+        /// <param name="listOfAllUniqueBlockHeaders">List of All unique blockheader in all connected chains.</param>
+        private void CheckChainedHeaderTreeConsistency(ChainedHeaderTree cht, IEnumerable<BlockHeader> listOfAllUniqueBlockHeaders)
+        {
+            bool noUnclaimedBraches = true;
+            bool eachPeerOneEntry = true;
+            bool reflecttipHash = true;
+            bool chbhHasOnlyReachebleHeaders = true;
+            bool isCorrectChainSequence = true;
+            bool hasLocalPeer = false;
+
+            var peerEntryDictionary = new Dictionary<int, int>();
+            var peerHDictionary = new Dictionary<int, int>();
+
+            Dictionary<uint256, HashSet<int>> tipsDictionary = cht.GetPeerIdsByTipHash();
+            Dictionary<int, uint256> peerIdsDictionary = cht.GetPeerTipsByPeerId();
+
+            peerHDictionary = new Dictionary<int, int>();
+            for (int i = 0; i < peerIdsDictionary.Count - 1; i++)
+            {
+                peerHDictionary.Add(i, 0);
+                peerEntryDictionary.Add(i, 0);
+            }
+
+            foreach (KeyValuePair<uint256, HashSet<int>> tips in tipsDictionary)
+            {
+                if (tips.Value.Count == 0) noUnclaimedBraches = false;
+
+                foreach (int peerId in tips.Value)
+                {
+                    // Ignore local.
+                    if (peerId != -1) 
```

Comment:
There is a constant for the local PID.

Reply:
yes but it's internal, so cannot be used in instance.
 internal const int LocalPeerId = -1;

Reply:
can - https://github.com/kogot/StratisBitcoinFullNode/blob/cb66f8717a703b05499597c100f7f1dfa64f88dd/src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs#L305